src/support.cpp

```diff
@@ -1413,7 +1398,30 @@ void AreaSupport::handleTowers(const Settings& settings, Polygons& supportLayer_
 
         if (tower_roof[0].area() < tower_diameter * tower_diameter)
         {
-            tower_roof = tower_roof.offset(tower_roof_expansion_distance);
+            const bool no_support = false;
+            const bool no_prime_tower = false;
+            Polygons model_outline = storage.getLayerOutlines(layer_idx, no_support, no_prime_tower);
+
+            // Rather than offsetting the tower with tower_roof_expansion_distance we do this step wise to achieve two things
+            // - prevent support from folding around the model
+            // - provide method to early out the offsetting procedure when the desired area is reached
+            const coord_t offset_per_step = support_line_width / 2;
+            for (coord_t offset_cumulative = 0; offset_cumulative <= tower_roof_expansion_distance; offset_cumulative += offset_per_step)
```

Comment:
does this have to be such an ancient for loop?
Feels like this can be done with a range